server/schema/user/user.datasources.js

```diff
@@ -178,19 +178,31 @@ const updateDetails = (id, fields, session, authToken, mid) =>
     { new: true }
   );
 
-const updateRoles = async (id, roles) => {
+const getCustomClaims = async (id) => {
   try {
     const _user = await UserModel.findById(id, 'email');
     if (!_user) {
       throw APIError('NOT_FOUND', null, { reason: 'The requested user does not exist.' });
     }
+    const _fbUser = await admin.auth().getUserByEmail(_user.email);
+    const userRecord = await admin.auth().getUser(_fbUser.uid);
+    return userRecord.customClaims;
+  } catch (error) {
+    throw FirebaseAuthError(error, { reason: "Cannot find user's roles" });
+  }
+};
+const updateRoles = async (id, roles) => {
```

Comment:
Since the upper datasource is named as getCustomClaims, allow this one to also be updateCustomClaims, which does not only update roles but updates the whole dataset.